daffodil-core/src/main/scala/org/apache/daffodil/dsom/SimpleTypes.scala

```diff
@@ -568,7 +568,11 @@ final class EnumerationDef(
   override lazy val optRepType = parentType.optRepType
 
   lazy val enumValueRaw: String = (xml \ "@value").head.text
-  lazy val enumValueCooked: DataValuePrimitive = parentType.primType.fromXMLString(enumValueRaw)
+  lazy val enumValueCooked: DataValuePrimitive = try {
+    parentType.primType.fromXMLString(enumValueRaw)
+  } catch {
+    case e: Throwable => SDE("Invalid data for enumeration: %s".format(e.getMessage))
```

Comment:
Dont catch throwable. You have good knowledge of the types that can be thrown here. 
SDE takes a format string and varargs. Don't call format yourself.